src/assets.rs

```diff
@@ -183,42 +178,42 @@ impl<'a> AssetEditor<'a> {
     ///
     /// Returns the stored asset.
     pub fn save<L: Ledger>(&mut self) -> Result<Asset, KeystoreError<L>> {
-        self.update();
+        if let Ok(store_asset) = self.store.load(&self.asset.definition.code) {
+            self.update(store_asset);
+        }
```

Comment:
I don't think we should actually do this here. For one thing, `update` is asymmetric, so `self.update(store_asset)` will overwrite changes to `self` with the existing contents of `store_asset`. But even if we did it in the other order, `update` will always take `Some` when one of a pair of corresponding fields is `Some`, which makes, e.g. `clear_icon` a noop. I believe `update` should instead be called in `create`, so that we don't overwrite an existing asset with default metadata when calling `create`